### PR TITLE
Update rules_go and use added race-detection feature

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR moves Mixer to use the most recent version of `rules_go`. One of the motivations for moving is the added ability to use the `--features=race` flag for testing, which enables the elimination of the current redundant test runs done for race checking.

This also keeps Mixer on-pace with Pilot, and the recent PR to upgrade their `rules_go` dep as well. This includes ensuring builds use static linking mode. For ref: https://github.com/istio/pilot/pull/958.

Of course, moving to a 0.5.x version of `rules_go` crosses deprecation boundaries. In particular, `new_go_repository` is no longer supported (`go_repository` is the new hotness).

Part of this PR removes the `gogo_proto_repositories` initialization call in the WORKSPACE, opting instead for a direct inclusion of deps that that rule was importing.

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/951)

### CODE REVIEW

File: bin/pre-push [hunk 1]
```diff
@@ -28,6 +28,6 @@ if ! [[ -f WORKSPACE ]]; then
   exit 1
 fi
 
-bazel test //...
-bin/racetest.sh
+bazel test --features=race //...
+# bin/racetest.sh
```
Comment (mandarjog) [inline bin/pre-push]: remove?

File: tools/codegen/pkg/bootstrapgen/generator.go [hunk 2]
```diff
@@ -32,16 +28,16 @@ func (g *Generator) Generate(_ []string) error {
...
-	if err != nil {
-		return err
-	}
-	defer func() { _ = f1.Close() }()
-	if _, err = f1.Write([]byte{}); err != nil {
-		_ = f1.Close()
-		_ = os.Remove(f1.Name())
-		return err
-	}
+	//f1, err := os.Create(g.OutFilePath)
```
Comment (mandarjog) [inline tools/codegen/pkg/bootstrapgen/generator.go]: delete commented code ?

File: tools/bazel.rc [hunk 3]
```diff
@@ -3,3 +3,5 @@ test --test_size_filters=-large,-enormous
 
 # Always include the version information in the build
 build --workspace_status_command=./bin/build_stamp.sh --action_env=ISTIO_VERSION
+
```
Comment (mandarjog) [inline tools/bazel.rc]: comment about why this option is used.

File: tools/codegen/pkg/bootstrapgen/generator.go (hunk 2 above)
Comment (douglas-reid) [inline tools/codegen/pkg/bootstrapgen/generator.go]: this was in-progress code with a TODO. felt better to not delete, but still needed to do something to make linters happy.

File: bin/pre-push (hunk 1 above)
Comment (douglas-reid) [inline bin/pre-push]: done.

File: tools/bazel.rc (hunk 3 above)
Comment (douglas-reid) [inline tools/bazel.rc]: done.

**Comment by douglas-reid** [comment]
build mixer/e2e-smoketest

**Comment by douglas-reid** [comment]
build mixer/e2e-smoktest

/sigh run it once, the mixer tests pass, bookinfo fails. run it again, mixer fails, bookinfo passes. completely environment dependent.

**Comment by douglas-reid** [comment]
build mixer/e2e-smoketest

**Comment by douglas-reid** [comment]
build mixer/e2e-smoketest

**Pull request merged**